Added optional scale line to prints

VulpesFerrilata opened the pull request:
This PR adds an optional scale line or bar when exporting to a PDF for printing.

There is a new checkbox in the Export and Print panel. When ticked a scale line or bar (depending on which one the user has active) will be added to the export.

A scale line/bar will be added to the bottom-right of the PDF export, above the attributions. The position and scale will adapt depending on the size (A4, A3 etc), orientation and the print resolution (150 dpi, 200 dpi etc).
![image](https://github.com/user-attachments/assets/bc0bcecb-467b-48d5-9e79-eedb6da58ad0)

Observation: the scale bar changes in size when you swap between different dpis - appearing bigger at lower dpis and smaller at higher dpis. I think this is to do with how OpenLayers defines the size of a pixel, and the same scaling issue happens with measurements and annotations.

On staging for full testing.

Closes #292

RobQuincey-DC commented:
Found an issue with the old school scale bar 
![image](https://github.com/user-attachments/assets/5ebbd1a5-522f-4b62-ad26-dde88ef2aa27)
It seems to be clipping. I'm guessing this is something to do with html2canvas, but I haven't looked closer yet

VulpesFerrilata commented:
Cheers @RobQuincey-DC  - the clipping is now fixed. I've also added some comments for clarity of some of the quirks we discovered while doing this feature and got rid of a couple of magic numbers.

RobQuincey-DC commented:
I've made two nitpick changes.

I've moved the one inch constant to be a part of the class, that way we only define it once, and use it again and again in the rest of the code. I did the same with the 96 DPI resolution that pops up a few times.

I also tweaked the logic of the code that decides the Y position of the scale line based on the DPI of the print. It now uses `<=` rather than `=` to allow of the DPIs in between the ones we've already defined, just in case we change them or add different ones at a future point.
